### DISCUSSION TITLE
TS011F_din_smart_relay NOT supported Zigbee2MQTT 1.25.0

### DISCUSSION THREAD

**Discussion by erwan33fr** [Koenkk/zigbee2mqtt#12051] [Devices]
Hello,

I have by this device on Aliexpress, ATMS1602Z :
https://fr.aliexpress.com/item/1005003473545843.html?gatewayAdapt=glo2fra&spm=a2g0o.order_list.0.0.21ef5e5bIm1q6a

![TS011F_din_smart_relay](https://user-images.githubusercontent.com/86933399/174989333-0fa04b74-5942-4bf4-bc7a-aa14fabfb1a4.jpg)

It seems to be part of the supported device :
https://www.zigbee2mqtt.io/devices/TS011F_din_smart_relay.html

When i add the device I have this error message :
Device '0xaXXXXXXXXXXXXXXXXX' with Zigbee model 'TS011F' and manufacturer name '_TZ3000_ky0fq4ho' is NOT supported

It seems that the model name returned by the device is not the correct one
"TS011F"
When it should be : "TS011F_din_smart_relay"

I use Zigbee2MQTT version
[1.25.0](https://github.com/Koenkk/zigbee2mqtt/releases/tag/1.25.0) commit: [6f1460e4](https://github.com/Koenkk/zigbee2mqtt/commit/6f1460e4)

Coordinator type
zStack3x0
Coordinator revision
20210708

Total 23
By device type
Router: 16
End devices: 7
By power source
Mains (single phase): 16
Battery: 7

**Comment by Koenkk** [comment]
`_TZ3000_ky0fq4ho` was not supported yet, added it now.

Changes will be available in the dev branch in a few hours from now. (https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html)

**Reply by erwan33fr** [reply]
I switched to the dev branch and the device is well supported.
Thank you very much for your response.

**Comment by kondrikserg** [comment]
i have a similar problem. 
https://aliexpress.ru/item/1005003456213043.html?spm=a2g39.orderlist.0.0.1f204aa6uLN96s
I have a device: _TZ3000_7issjl2q. Model name: ATMS1601Z. Is it possible to enable support for it?

**Reply by kondrikserg** [reply]
I can't connect my device _TZ3000_7issjl2q :(

**Reply by prokudin07** [reply]
> I can't connect my device _TZ3000_7issjl2q 
@kondrikserg 
It's some other device. It's without power monitoring. I added this device in my fork. I think after 1-2 day I'll do pull request and device will able in dev branch.

Model ATMS1602Z  has got power monitoring
Model ATMS1601Z  has not got power monitoring.

**Comment by prokudin07** [comment]
I created PR [4203](https://github.com/Koenkk/zigbee-herdsman-converters/pull/4203)

**Reply by erwan33fr** [reply]
Is it possible to modify the code for ATMS1602Z to allow the configuration of the indicator LED to be modified as it is possible to do under "Smart Life"?
https://github.com/Koenkk/zigbee2mqtt/discussions/12244

**Reply by prokudin07** [reply]
I tried to do exactly this feature for few hours, but can not to do this... :( 
Very want to do this, but....

**Reply by erwan33fr** [reply]
ATMS1602Z probably needs to be updated to the latest Firmware via "Smart Life" first to be able to change the indicator LED settings via Zigbee2mqtt. (apparently the possibility of modifying the state of the indicator led was added in the last version)

**Reply by prokudin07** [reply]
No. I see code in Z2M and it's not there.  This option is not there at all.

**Reply by erwan33fr** [reply]
However, this option are present via the "smart life" application, so there should be zigbee commands to modify it ?

**Comment by prokudin07** [comment]
Yes, you are absolutely right, but I can not to do this.

**Comment by prokudin07** [comment]
@erwan33fr 
I did it!!!! 

https://user-images.githubusercontent.com/81035252/166991129-802364ea-ff02-40a9-9910-2d45775e9ca6.mp4

Later will do PR

**Reply by erwan33fr** [reply]
Great :-))))
I would be curious to know how to modify the code of a device under zigbee2mqtt as you did.
Your modification will be imported in the dev branch shortly ?

**Reply by prokudin07** [reply]
This is code for reading status of led:
```
    tuya_relay_din_led_indicator: {
        cluster: 'genOnOff',
        type: ['attributeReport', 'readResponse'],
        convert: (model, msg, publish, options, meta) => {
            const property = 0x8001;

            if (msg.data.hasOwnProperty(property)) {
                const dict = {0x00: 'off', 0x01: 'on_off', 0x02: 'off_on'};
                const value = msg.data[property];

                if (dict.hasOwnProperty(value)) {
                    return {[postfixWithEndpointName('indicator_mode', msg, model)]: dict[value]};
                }
            }
        },
    },
```
search, direction from my friend, try 1, try 2 ...  try N and got it ))

I hope about hour.. Then Koenkk

**Reply by prokudin07** [reply]
> Great :-)))) I would be curious to know how to modify the code of a device under zigbee2mqtt as you did. Your modification will be imported in the dev branch shortly ?
@erwan33fr 
I did [4213](https://github.com/Koenkk/zigbee-herdsman-converters/pull/4213)

**Reply by erwan33fr** [reply]
@prokudin07 
Perfect :-)
I'm starting to understand, the development and validation process :-)

**Reply by prokudin07** [reply]
> @prokudin07 
> Perfect :-)
> I'm starting to understand, the development and validation process :-)

Hi. Pull request was done. Can you check your device in Z2M Edge?

**Reply by erwan33fr** [reply]
Perfect, there is now an "indicator_mode" attribute which works :-)
This is exactly what I was looking for. Thanks very much :-)
Just a remark, I don't know if it's normal, and it's a detail, in Home Assistant, this new field appears in "Controls" and not in "Configuration" (like "power_outage_memory" for example)

**Reply by prokudin07** [reply]
I think it's a feature HA

**Comment by andrey170183** [comment]
The TS011F_din_smart_relay device is normally determined. Switching on /off works and measures the voltage, but does not show instantaneous power....
Writes in the logs:
No converter available for 'TS011F_din_smart_relay' with cluster '57344' and type 'raw' and data '{"data":[8,63,10,1,208,72,2,0,6],"type":"Buffer"}'

Help solve the problem.

**Reply by erwan33fr** [reply]
For the device with a high cos φ and low consumption (like led bulb), the power measurement does not work. For this kind of device I rather use the current measurement.
Can you try with another device, like an incandescent bulb to test?

**Reply by andrey170183** [reply]
Yes, I connected other devices with a power of more than 1000W (kettle) to it. It still shows zero.

**Reply by erwan33fr** [reply]
Indeed, it is strange
Are the other measurements, voltage and current correct ?

**Reply by andrey170183** [reply]
Yes, here is a screenshot.
![image](https://user-images.githubusercontent.com/42129962/171346831-6db552c4-ceda-49c7-8b22-2c59ce375c12.png)

**Reply by erwan33fr** [reply]
ATMS1602Z or ATMS1601Z ?

**Reply by erwan33fr** [reply]
The power cannot be correct since the current is not correctly measured. Do you have the electrical wiring ?
Are the phase and neutral of the equipment connected in the correct place?

**Reply by andrey170183** [reply]
Since the neutral of the incoming and outgoing are connected to each other, I have put together such a scheme
![image](https://user-images.githubusercontent.com/42129962/171350561-0b97cd5a-a217-4f88-910e-faedcf112f45.png)


> ATMS1602Z or ATMS1601Z ?
Where i can see it?

**Reply by erwan33fr** [reply]
It's written on it

**Reply by erwan33fr** [reply]
With this wiring it's normal that it doesn't work ;-)
This is not a Zigbee2mqtt problem, but an electrical wiring problem.

**Reply by andrey170183** [reply]
I measured the tester between the neutrals. I thought it was short-connected there))

**Reply by erwan33fr** [reply]
There is indeed continuity between the two neutrals, but with an integrated ammeter between the two ;-)

**Comment by erwan33fr** [comment]
I noticed something strange with this TS011F_din_smart_relay "ATMS1602Z".

When the device is in the ON state, approximately every 5 minutes, the current measurement then the power measurement change to the value 0 for approximately 5 seconds. Then the measurement returns to normal.

![Log KO](https://user-images.githubusercontent.com/86933399/174988083-b5ce1d81-35f6-4037-a57c-f2f79851b453.JPG)


I also noticed that sometimes the device goes to OFF state by itself.

![Graph KO](https://user-images.githubusercontent.com/86933399/174988018-fe3eecaf-8ed0-4e32-8fa6-8c5507d52c6f.jpg)

Has anyone had these symptoms before?

**Reply by erwan33fr** [reply]
The switch goes to the "OFF" state on its own have been solved by an update of the device via "Smart Life" application :
1.0.10 => 1.0.13

There is always the measurement problem at 0 every 5 minutes :-(

**Reply by erwan33fr** [reply]
I noticed that the passage of the current measurement to 0 was about every 5 minutes and more precisely every 0.06 Kwh
